File: resource-server/src/main/java/by/dima/resourceserver/service/DocumentService.java
package by.dima.resourceserver.service;

import by.dima.resourceserver.model.Document;
import by.dima.resourceserver.model.User;

import java.util.Collection;
import java.util.List;

public interface DocumentService {
    void save(Document document, User user);
    void delete(Long id, User user);
    void archive(Long id, User user);
    void sign(Long id, User user);

    Document get(Long id, User user);
    Document get(String name, User user);

    List<Document> getAll(String username);
    List<Document> getActive(User user);
    List<Document> getArchived(User user);
    Collection<Document> getSign(User user);
}

File: client-app/src/app/service/document.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {UserService} from './user.service';
import {Document} from '../model/document';
import {Observable} from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class DocumentService {

  constructor(private http: HttpClient, private userService: UserService) {
  }

  userList(): Observable<string[]> {
    return this.userService.authGet('http://localhost:8080/signs/users');
  }

  create(document: Document): Observable<any> {
    return this.userService.authPost('http://localhost:8080/documents/', document);
  }

  archive(document: Document): Observable<any> {
    return this.userService.authPost('http://localhost:8080/archive/' + document.id);
  }

  sign(document: Document): Observable<any> {
    return this.userService.authPost('http://localhost:8080/signs/' + document.id);
  }

  delete(document: Document): Observable<any> {
    return this.userService.authDelete('http://localhost:8080/documents/' + document.id);
  }

  getMy(): Observable<Document[]> {
    return this.userService.authGet<Document[]>('http://localhost:8080/documents/me');
  }

  getSign(): Observable<Document[]> {
    return this.userService.authGet<Document[]>('http://localhost:8080/signs');
  }

  getActive(): Observable<Document[]> {
    return this.userService.authGet<Document[]>('http://localhost:8080/documents/active');
  }

  getArchived(): Observable<Document[]> {
    return this.userService.authGet<Document[]>('http://localhost:8080/archive');
  }
}
